github-runner-provisioner/github.go

```diff
@@ -3,20 +3,65 @@ package main
 import (
 	"context"
 	"github.com/google/go-github/v48/github"
+	"golang.org/x/exp/slices"
 	"golang.org/x/oauth2"
 )
 
-func getGitHubRunnerToken(ctx context.Context, owner string, repo string) (token string, err error) {
+func getGitHubAPIClient(ctx context.Context) *github.Client {
 	ts := oauth2.StaticTokenSource(
 		&oauth2.Token{AccessToken: config.GithubToken},
 	)
 	tc := oauth2.NewClient(ctx, ts)
-
 	client := github.NewClient(tc)
+	return client
+}
+
+func getGitHubRunnerToken(ctx context.Context, owner string, repo string) (token string, err error) {
+	client := getGitHubAPIClient(ctx)
 	registrationToken, _, err := client.Actions.CreateRegistrationToken(ctx, owner, repo)
 	if err != nil {
 		return "", err
 	}
 
 	return *registrationToken.Token, nil
 }
+
+func getGitHubRunners(ctx context.Context, owner string, repo string) *github.Runners {
+	client := getGitHubAPIClient(ctx)
+	opts := &github.ListOptions{}
+	runners, _, err := client.Actions.ListRunners(ctx, owner, repo, opts)
+	if err != nil {
+		return nil
+	}
+	return runners
+}
+
+func isRunnerAvailable(ctx context.Context, owner string, repo string, labels []string) bool {
+	runners := getGitHubRunners(ctx, owner, repo)
+
+	//check all runners registered to the repo
+	for i := range runners.Runners {
```

Comment:
This can cause a null pointer exception since `getGitHubRunners` can return nil.